### DISCUSSION TITLE
Tailwind’s purging mechanism eliminates inline classes during the build process.

### DISCUSSION THREAD

**Discussion by KonstantinosVazaios** [rewindui/rewindui#125] [General]
Yesterday I read about **Tailwind’s purging mechanism** that eliminates inline classes during the build process.
I forked rewindui component library, I created some new components as well with inline classNames (not using ts files with cva) and I was playing around with rollup.js and tried to build the library. After the library was build I installed it in my ReactJS frontend project, imported the styles inside the tailwind.config.js as mentioned and used storybook to create some stories to check the components I had created. 
I noticed that the Card component that I have made was almost fully broken and no styles were attached to it even though the were valid tailwind classes!
After doing my research I found this:

_TailwindCSS scans your files for class names to include in the final CSS output. If a class name is dynamically generated or inline in a compiled library, Tailwind won't see it during the build process of the user’s project._

I search the whole rewindui folder for components that had directly className="..." and you can see for example at rewindui storybook website inside the Checkbox component there is an "inline" className "grid-cols-1" that simply does nothing. Even though it should apply this style "grid-template-columns: repeat(1, minmax(0, 1fr));"

**_Tailwind’s purging mechanism_**

**Comment by KonstantinosVazaios** [comment]
A solution for this is basically change tailwind config file το './node_modules/@rewind-ui/core/dist/**' instead of importing the styles

**Comment by NickDunas** [comment]
This kind of got me confused. Isn't this what I am suggesting in the introduction in the docs?

**Reply by KonstantinosVazaios** [reply]
Not really, in the introduction docs this line of code is mentioned to be added:
'./node_modules/@rewind-ui/core/dist/theme/styles/*.js',

This instruct tailwind to scan only the styles, however inside rewind-ui there are components that use inline className like the Checkbox component.

I am also providing a screenshot of rewind's ui Checkbox story, so you can see that the grid-cols-1 do not take any effect (https://storybook.rewind-ui.dev/?path=/docs/components-checkbox--docs)
![image](https://github.com/user-attachments/assets/8fd8567d-9959-45a2-862e-739369f30635)

**Reply by NickDunas** [reply]
Oh now I get it!! Thank you!! I will try to figure out a proper way to fix that! :)

**Reply by KonstantinosVazaios** [reply]
You are welcome Nick!

I am thinking if setting up a tailwind config file inside the rewindui component library would make sense... just like other component libraries do like shadcn.

With proper configuration the end user that will use the rewindui library, will not be forced to import styles from the node_modules which is a little bit "awkward". It will play out of the box
